File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int prod = 1;
        int count = 0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==1){
                count++;
            }
            prod = prod*nums[i];
        }

        int with_zero_prod = 1;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0) continue;

            with_zero_prod = nums[i] * with_zero_prod;
        }

        int arr[] = new int[nums.length];

        for(int i=0;i<nums.length;i++){
            if(nums[i]!=0) {
                arr[i] = prod/nums[i];
            }

            else if(nums[i]==0 && count==1){
                arr[i] = with_zero_prod;
            }

            else if(nums[i]==0 && count>1){
                arr[i] = 0;
            }
        }

        return arr;
    }
}  

File: Data Structures & Algorithms/find-all-numbers-disappeared-in-an-array/submission-0.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        seen = set()
        for i in range(len(nums)):
            seen.add(nums[i])
        
        l = []

        for i in range(1,len(nums)+1):
            if i in seen:
                continue
            else:
                l.append(i)
        
        return l